src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java

```diff
@@ -1925,6 +1924,76 @@ public void typeThemeTrainerPokes(Settings settings) {
         this.setTrainers(currentTrainers, false);
     }
 
+    @Override
+    public void randomizeTrainerHeldItems(Settings settings) {
+        boolean giveToBossPokemon = settings.isRandomizeHeldItemsForBossTrainerPokemon();
+        boolean giveToImportantPokemon = settings.isRandomizeHeldItemsForBossTrainerPokemon();
+        boolean giveToRegularPokemon = settings.isRandomizeHeldItemsForBossTrainerPokemon();
+        boolean highestLevelOnly = settings.isHighestLevelGetsItemsForTrainers();
+        List<Trainer> currentTrainers = this.getTrainers();
+        for (Trainer t : currentTrainers) {
+            if (!giveToRegularPokemon && (!t.isImportant() && !t.isBoss())) {
+                continue;
+            }
+            if (!giveToImportantPokemon && t.isImportant()) {
+                continue;
+            }
+            if (!giveToBossPokemon && t.isBoss()) {
+                continue;
+            }
+
+            if (highestLevelOnly) {
+                int maxLevel = -1;
+                TrainerPokemon highestLevelPoke = null;
+                for (TrainerPokemon tp : t.pokemon) {
+                    if (tp.level > maxLevel) {
+                        highestLevelPoke = tp;
+                        maxLevel = tp.level;
+                    }
+                    tp.heldItem = 0; // set every poke to no item (just in case there was one originally).
+                }
+                if (highestLevelPoke == null) {
+                    continue; // should never happen - trainer had zero pokes
+                }
+                randomizeHeldItem(highestLevelPoke, settings);
+            } else {
+                for (TrainerPokemon tp : t.pokemon) {
+                    randomizeHeldItem(tp, settings);
+                }
+            }
+        }
+        this.setTrainers(currentTrainers, false);
+    }
+
+    private void randomizeHeldItem(TrainerPokemon tp, Settings settings) {
+        boolean sensibleItemsOnly = settings.isSensibleItemsOnlyForTrainers();
+        boolean consumableItemsOnly = settings.isConsumableItemsOnlyForTrainers();
+        List<Integer> toChooseFrom;
+        if (sensibleItemsOnly) {
+            toChooseFrom = getSensibleHeldItemsFor(tp, consumableItemsOnly);
+        } else if (consumableItemsOnly) {
+            toChooseFrom = getAllConsumableHeldItems();
+        } else {
+            toChooseFrom = getAllHeldItems();
+        }
+        tp.heldItem = toChooseFrom.get(random.nextInt(toChooseFrom.size()));
+    }
+
+    @Override

```

Comment:
There's a section *way* at the bottom of AbstractRomHandler.java for "Default Implementations" as the comment calls it. Can you move these three methods here?

Reply:
Done.